services/fluxmonitor_test.go

```diff
@@ -164,62 +168,233 @@ func TestNewFluxMonitor(t *testing.T) {
 			0,
 			"50500",
 		},
+		{
+			"3 adapters, no threshold",
+			[]string{"50500", "51000", "50000"},
+			0,
+			0,
+			15 * time.Millisecond,
+			3 * time.Millisecond,
+			"50500",
+		},
+		{
+			"3 adapters, 1 non-expected response",
+			[]string{"50000", "51000", "wrong"},
+			0.01,
+			600,
+			15 * time.Millisecond,
+			3 * time.Millisecond,
+			"50500",
+		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			var mockAdapters []url.URL
-			for _, v := range tt.adapterResults {
-				mockAdapters = append(mockAdapters, *createMockAdapter(v))
-			}
+		var mockAdapters []url.URL
+		for _, v := range tt.adapterResults {
+			mockAdapters = append(mockAdapters, *createMockAdapter(v))
+		}
 
-			triggerJobRun := make(chan subscriber.Event, 10)
-			var fmConfig FluxMonitorConfig
+		triggerJobRun := make(chan subscriber.Event, 10)
+		var fmConfig FluxMonitorConfig
 
-			fmConfig.Adapters = mockAdapters
-			fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
-			fmConfig.Multiply = 18
-			fmConfig.Threshold = tt.threshold
-			fmConfig.AbsoluteThreshold = tt.absoluteThreshold
-			fmConfig.Heartbeat = tt.heartbeat
-			fmConfig.PollInterval = tt.heartbeat
-			fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 1 * time.Second, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 1 * time.Second}
+		fmConfig.Adapters = mockAdapters
+		fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
+		// fmConfig.Multiply = tt.multiply
+		fmConfig.Threshold = tt.threshold
+		fmConfig.AbsoluteThreshold = tt.absoluteThreshold
+		fmConfig.Heartbeat = tt.heartbeat
+		fmConfig.PollInterval = tt.heartbeat
+		fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 100 * time.Millisecond, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 10 * time.Millisecond}
 
+		t.Run("1 new round event tests: "+tt.name, func(t *testing.T) {
+			fmt.Printf("Testing %s", t.Name())
```

Comment:
In all of these tests we are printing the test name right after initiating the test with a name, so this seems unnecessary unless I'm missing some use for it.

Reply:
Added this because there is no indicator in logs of which test is currently running. Any thoughts?

Reply:
Alright, so it looks like you might be running `go test ./..` or something? If you add the -v flag, it should give you the test names :wink:

Reply:
Ah, sweet. Removing these.